### PR TITLE
Tasks skeleton and tests for tasks 1.1.1 to 1.1.5

### PR DESCRIPTION
**Pull request by STeeL835**
### Overview

- Created a new solution for tasks (felt that the previous one was a POC that was only for testing)
- Created tasks/tests for 
  - [ ] 1.1.1
  - [ ] 1.1.2
  - [ ] 1.1.3
  - [ ] 1.1.4
  - [ ] 1.1.5 (changed the task to accept not only a 1000 as a max number)

### Action items

- [ ] All todo comments are resolved
- [ ] The test pipeline should work
- [ ] Do we need tests for incorrect values in these tasks (except for 1, which explicitly asks for a check)

### CODE REVIEW

**Comment by STeeL835** [comment]
3 questions:
- is it okay that I created a new solution?
- Are we gonna use NUnit or XUnit?
- For mocks are we gonna use Moq or NSubstitute?

I've used NUnit and Moq already, but I can rewrite it, while there's not much.

**Comment by MarzipanCoin** [comment]
> 3 questions:
> 
> * is it okay that I created a new solution?
> * Are we gonna use NUnit or XUnit?
> * For mocks are we gonna use Moq or NSubstitute?
> 
> I've used NUnit and Moq already, but I can rewrite it, while there's not much.

You can use, what you want, but you need check CI pipeline. If CI has a problem then you need fix CI config file

File: test/Epam.ExternalTraining.Task1.Tests/Task1.1/TriangleTaskTests.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+			_triangleTask = new TriangleTask();
+		}
+
+		#region Test Cases
+
+		public static IEnumerable<TestCaseData> Run_RegularNumbers_ShouldDrawTheTriangle_TestCases()
+		{
+			yield return new TestCaseData(1, "*");
+			yield return new TestCaseData(3, AdjustLiteralStringForTests(@"
+*
```
Comment (Marzipan-Coin) [inline test/Epam.ExternalTraining.Task1.Tests/Task1.1/TriangleTaskTests.cs:29]: Could you extract test data to another class with string constants?

File: test/Epam.ExternalTraining.Task1.Tests/Task1.1/XmasTreeTaskTests.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+			_xmasTreeTask = new XmasTreeTask();
+		}
+
+		#region Test Cases
+
+		public static IEnumerable<TestCaseData> Run_RegularNumbers_ShouldDrawTheXmasTree_TestCases()
+		{
+			yield return new TestCaseData(1, "*");
+			yield return new TestCaseData(3, AdjustLiteralStringForTests(@"
+  *
```
Comment (Marzipan-Coin) [inline test/Epam.ExternalTraining.Task1.Tests/Task1.1/XmasTreeTaskTests.cs:26]: Could you extract test data to another class with string constants?

File: test/Epam.ExternalTraining.Task1.Tests/Task1.1/AnotherTriangleTaskTests.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+		public static IEnumerable<TestCaseData> Run_RegularNumbers_ShouldDrawTheIsoscelesTriangle_TestCases()
+		{
+			yield return new TestCaseData(1, "*");
+			yield return new TestCaseData(3, AdjustLiteralStringForTests(@"
+  *
+ ***
+*****"));
+			yield return new TestCaseData(15, AdjustLiteralStringForTests(@"
+              *
```
Comment (Marzipan-Coin) [inline test/Epam.ExternalTraining.Task1.Tests/Task1.1/AnotherTriangleTaskTests.cs:30]: Could you extract test data to another class with string constants?

File: test/Epam.ExternalTraining.Task1.Tests/Task1.1/AnotherTriangleTaskTests.cs [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+using FluentAssertions;
+using NUnit.Framework;
+using System.Collections.Generic;
+
+namespace Epam.ExternalTraining.Task1.Tests
+{
+	/// <summary> Task 1.1.3 - Another triangle (isosceles triangle) </summary>
+	public class AnotherTriangleTaskTests : ConsoleTestBase
+	{
+		private IAnotherTriangleTask _anotherTriangleTask;
```
Comment (Marzipan-Coin) [inline test/Epam.ExternalTraining.Task1.Tests/Task1.1/AnotherTriangleTaskTests.cs:12]: IAnotherTriangleTask  _blaBlatask;

        [OneTimeSetUp]
        public void Initialize()
        {
            if (_painter == null)
            {
                _blaBlaTask= Assembly.GetAssembly(
                typeof(IAnotherTriangleTask  ))
                .GetTypes()
                .Where(t => t.GetInterfaces()
                .Contains(typeof(IAnotherTriangleTask  )))
                .Select(t => Activator.CreateInstance(t) as IAnotherTriangleTask  )
                .FirstOrDefault();
            }
        }

It's need for providing for students only interfaces without implementation

Comment (STeeL835) [reply test/Epam.ExternalTraining.Task1.Tests/Task1.1/AnotherTriangleTaskTests.cs:12]: @Marzipan-Coin I think we can leave a stub class that implements the interface with `NotImplementedException`s. No reflection and no problem solution, and the difference of having an empty class already isn't that big.

Comment (Marzipan-Coin) [reply test/Epam.ExternalTraining.Task1.Tests/Task1.1/AnotherTriangleTaskTests.cs:12]: Wow, Nice idea

**Comment by STeeL835** [comment]
Moved to https://github.com/ElioWrite/epam_xt_selfcheck/pull/1

**Pull request closed without merging**